src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -185,6 +185,24 @@ public void start(Map<String, String> properties) {
         }
       }
 
+      Long incrementingOffset = config.getLong(JdbcSourceConnectorConfig.INCREMENTING_OFFSET);
+      Long timestampOffset = config.getLong(JdbcSourceConnectorConfig.TIMESTAMP_OFFSET);
+
+
+      if (incrementingOffset != null) {
+        if (offset == null) {
+          offset = new HashMap<>();
+        }
+        offset.put("incrementing", incrementingOffset);
+      }
+
+      if (timestampOffset != null) {
+        if (offset == null) {
+          offset = new HashMap<>();
+        }
+        offset.put("timestamp", timestampOffset);
+      }
+
```

Comment:
Won't this always set the offset values to what is read from the config? So, if the connector task restarts (which will happen during a rebalance), if the `incrementing.offset` and/or `timestamp.offset` config properties are still set, then these lines will always use those values to update the offsets, and the connector task will essentially back up to this point. If it restarts again, it will again back up to this same exact point.

Also, the supplied `incrementing.offset` and `timestamp.offset` configuration values will be applied to all tables. Is that really what you want?